### DISCUSSION TITLE
lgtm integration

### DISCUSSION THREAD

**Discussion by shumonsharif** [quarkiverse/quarkus-cxf#390] [Ideas]
@gastaldi I've been thinking of introducing [lgtm](https://lgtm.com/) for the quarkus-cxf project, particularly for automatic code reviews for PRs.  Is this something you can help with, as I don't think I have access to the necessary GitHub settings?  Please do share your thoughts!

**Comment by gastaldi** [comment]
That looks interesting. I'd like to see how it works, are you using this service anywhere?

**Reply by gastaldi** [reply]
Also, how does it compare to Snyk? Would that be an option too?

**Reply by shumonsharif** [reply]
I'm not using the service, but I was looking for a free option when I came across it.   Here's another link for it:
https://github.com/marketplace/lgtm

I can try to set it up on a personal repo to experiment a little with it first, will keep you posted!

Snyk could also work, but it seems their free version has some limits - and those may possibly apply at the quarkiverse org level vs. project level.

**Reply by shumonsharif** [reply]
So, it was super simple to set up, seems to be working well so far ...
https://github.com/shumonsharif/rce/pull/2

EDIT: It took a good 6-8 mins to complete the analysis on the PR - not too bad - and the results are as expected, it successfully flagged the issue introduced in the linked PR.

**Reply by gastaldi** [reply]
I'm +1 to add it. WDYT @quarkiverse/quarkiverse-owners ?

**Reply by gastaldi** [reply]
Obviously, just as we did with the Stale app, that would be enabled on a per-repository basis: https://github.com/quarkiverse/quarkiverse-devops/blob/main/quarkus-rabbitmq-client.tf#L48-L54

**Reply by gastaldi** [reply]
I created a PR that enables the LGTM app for the `quarkus-cxf` repo: https://github.com/quarkiverse/quarkiverse-devops/pull/28

I'll wait for the team response before merging it

**Reply by gastaldi** [reply]
Just merged it, let's see how that works for you

**Reply by shumonsharif** [reply]
Thanks so much!  Look forward to trying it, we already had several issues flagged.
